component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java

```
@@ -248,7 +248,13 @@ private ActivityEntity convertCommentToCommentEntity(ExoSocialActivity comment)
     commentEntity.setTitleId(comment.getTitleId());
     commentEntity.setType(comment.getType());
     commentEntity.setBody(comment.getBody());
-    commentEntity.setPosterId(comment.getPosterId() != null ? comment.getPosterId() : comment.getUserId());
+    String result = comment.getPosterId() != null ? comment.getPosterId() : comment.getUserId();
+    if (result != null) {
+      commentEntity.setPosterId(result);
+    } else {
+      LOG.warn("Could not set POSTER_ID for comment " + comment.getId());
+      return null;
+    }
```

I don't think we should change the current behavior. It is not allowed to not have a poster, so it should throw an exception, as it is the case today. So it is up to the caller method to catch correctly the error and make sure the transaction is correctly ended.